client/src/language/codeActionProvider.ts

```diff
@@ -0,0 +1,101 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2023 Savoir-faire Linux. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import vscode from 'vscode'
+import { requestsManager } from './RequestManager'
+import { embeddedLanguageDocsManager } from './EmbeddedLanguageDocsManager'
+import { getEmbeddedLanguageDocRange, getIndentationOnLine, getOriginalDocPosition } from './utils'
+import { logger } from '../lib/src/utils/OutputLogger'
+import { type Range } from 'vscode-languageclient'
+
+export class BitbakeCodeActionProvider implements vscode.CodeActionProvider {
+  async provideCodeActions (document: vscode.TextDocument, range: vscode.Range | vscode.Selection, context: vscode.CodeActionContext, token: vscode.CancellationToken): Promise<vscode.CodeAction[]> {
+    const diagnostics = context.diagnostics
+    const actions = await Promise.all(
+      diagnostics.map(async (diagnostic) => await buildActionFromDiagnostic(document, diagnostic))
+    ).then((results) => results.flat())
+    return actions
+  }
+}
+
+const buildActionFromDiagnostic = async (document: vscode.TextDocument, diagnostic: vscode.Diagnostic): Promise<vscode.CodeAction[]> => {
+  const originalRange = diagnostic.range
+  const embeddedLanguageType = await requestsManager.getEmbeddedLanguageTypeOnPosition(document.uri.toString(), originalRange.start)
+  if (embeddedLanguageType === undefined || embeddedLanguageType === null) {
+    return []
+  }
+  const embeddedLanguageDocInfos = embeddedLanguageDocsManager.getEmbeddedLanguageDocInfos(document.uri.toString(), embeddedLanguageType)
+  if (embeddedLanguageDocInfos === undefined || embeddedLanguageDocInfos === null) {
+    return []
+  }
+  const embeddedLanguageTextDocument = await vscode.workspace.openTextDocument(embeddedLanguageDocInfos.uri)
+  const embeddedRange = getEmbeddedLanguageDocRange(
+    document,
+    embeddedLanguageTextDocument,
+    embeddedLanguageDocInfos.characterIndexes,
+    originalRange
+  )
+
+  const tempActions = await vscode.commands.executeCommand<vscode.CodeAction[]>(
+    'vscode.executeCodeActionProvider',
+    embeddedLanguageDocInfos.uri,
+    embeddedRange
+  )
+
+  const actions: vscode.CodeAction[] = []
+  tempActions.forEach((action) => {
+    switch (action.command?.command) {
+      case 'python.addImport':
+        handlePythonAddImport(action, document, embeddedLanguageTextDocument, embeddedLanguageDocInfos.characterIndexes)
+        break
+      default:
+        return
+    }
+    actions.push(action)
+  })
+
+  return actions
+}
+
+type PythonAddImportArguments = [
+  path: string,
+  range: Range,
+  targetName: string,
+  moduleName: string | null,
+]
+const handlePythonAddImport = (
```

Comment:
weird way to define a typescript function

Reply:
what do you mean? Looks quite normal to me

Reply:
Usually we use this in this codebase:
```Typescript
function handlePythonAddImport(args) : output {
    impl
}
```

Reply:
Well, the keyword `function` is found about 121 times in our codebase, while `=>` is there about 283 times.

I'm surprised our default eslint configuration does not have an opinion on this. Apparently we could set rules:
https://eslint.org/docs/latest/rules/func-style
https://eslint.org/docs/latest/rules/prefer-arrow-callback